Program.cs

```diff
@@ -0,0 +1,194 @@
+        public void Add(string part)
+        {
+            this.parts.Add(part);
+        }
+        public string ListParts()
+        {
+            return "White Pizza: " + string.Join(",", this.parts);
+        }
+    }
+    
+    public class WheatPizzaBuilder : IPizzaBuilder
+    {
+        private WheatPizza _product = new();
+        public WheatPizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+            this._product = new WheatPizza();
+        }
+        public void BuildDough(string doughType=null)
+        {
+            if(doughType == null ||string.IsNullOrEmpty(doughType) )
+            {
+                this._product.Add("Wheat Dough");
+            }
+            else
+            {
+                this._product.Add(doughType);
+            }
+        }
+        public void BuildSauce(string sauceType=null)
+        {
+            if(sauceType == null ||  string.IsNullOrEmpty(sauceType))
+            {
+                this._product.Add("Italian Sauce");
+            }
+            else
+            {
+                this._product.Add(sauceType);
+            }
+        }
+        public void BuildTopping(string toppingType=null)
+        {
+            if(toppingType == null ||  string.IsNullOrEmpty(toppingType))
+            {
+                this._product.Add("Cheese Marai Topping");
+            }
+            else
+            {
+                this._product.Add(toppingType);
+            }
+        }
+        public WheatPizza GetPizza()
+        {
+            WheatPizza result = this._product;
+            this.Reset();
+            return result;
+        }
+    }
+    public class WheatPizza
+    {
+        private List<object> parts = new List<object>();
+        public void Add(string part)
+        {
+            this.parts.Add(part);
+        }
+        public string ListParts()
+        {
+            return "Wheat Pizaa: " + string.Join(",", this.parts);
+        }
+    }
+    public class Director
+    {
+        private IPizzaBuilder _builder;
+
+        public Director(IPizzaBuilder pizzaBuilder)
+        {
+            this._builder = pizzaBuilder;
+        }
+        public void MakePizza()
+        {
+            WriteLine("Dough ? : Or tra Default");
+            this._builder.BuildDough( ReadLine());
+            WriteLine("Sauce ? : Or tra Default");
+            this._builder.BuildSauce( ReadLine());
+            WriteLine("Topping ?: Or tra Default ");
+            this._builder.BuildTopping( ReadLine());
+        }
+    }
+    class Program
+    {
+        static void Main()
+        {
+            //client
+            WhitePizzaBuilder whitePizzaBuilder = new();
+            Director director = new(whitePizzaBuilder);
+            // Bonus Two user prompt 
+            director.MakePizza();
+            WriteLine("------------------------------------");
+            WriteLine("Here IS your White Pizza ");
```

Comment:
You Shouldn't have written the pizza Type because you don't know what the user going to order, for example if the user entered a _brown pizza_ then you printed for them _Here IS your White Pizza_ they will be confused!